src/Redemption.js

```diff
@@ -165,21 +185,31 @@ export default class Redemption {
           )
         }
 
-        const requiredConfirmations = parseInt(
-          await this.deposit.factory.constantsContract.methods
-            .getTxProofDifficultyFactor()
-            .call()
-        )
+        return transaction
+      }
+    )
 
-        console.debug(
-          `Waiting for ${requiredConfirmations} confirmations for ` +
-            `Bitcoin transaction ${transaction.transactionID}...`
-        )
-        await BitcoinHelpers.Transaction.waitForConfirmations(
-          transaction,
-          requiredConfirmations
-        )
+    state.confirmations = state.signedTransaction.then(async transaction => {
+      const requiredConfirmations = parseInt(
+        await this.deposit.factory.constantsContract.methods
+          .getTxProofDifficultyFactor()
+          .call()
+      )
 
+      console.debug(
+        `Waiting for ${requiredConfirmations} confirmations for ` +
+          `Bitcoin transaction ${transaction.transactionID}...`
+      )
+      await BitcoinHelpers.Transaction.waitForConfirmations(
+        transaction,
```

Comment:
Heh. Really this should also take a `transactionID` rather than a `transaction`. Might be too deep a refactor right now though.

Reply:
Went for it. Updated!

Reply:
:sunglasses: